src/dbnode/x/xio/types.go

```diff
@@ -86,6 +86,10 @@ type ReaderSliceOfSlicesIterator interface {
 
 	// Size gives the size of bytes in this iterator.
 	Size() (int, error)
+
+	// Rewind returns the iterator to the beginning.
+	// This operation is invalid if any of the block readers have been read.
+	Rewind() error
```

Comment:
nit: can probably drop the error here, especially since it's an iterator we'd be calling `.Error()` on anyway